### PR TITLE
Docs: Fix Airbyte documentation supported version

### LINKED ISSUE #4398: Remove notice from Airbyte integration docs
Remove notice from Airbyte integration docs once https://github.com/airbytehq/airbyte/pull/18031 is merged and released.

- Reference: https://github.com/treeverse/lakeFS/pull/4397

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #4398

### CODE REVIEW

File: docs/integrations/airbyte.md [hunk 1]
```diff
@@ -30,7 +29,7 @@ The integration between the two open-source projects brings resilience and manag
 connectors to sync data to your S3 buckets by leveraging lakeFS branches and atomic commits and merges.
 
 ## Use cases
-You can take advantage of lakeFS consistency guarantees and [CI/CD capabilities](../using_lakefs/ci.md) when ingesting data to S3 using lakeFS:
+You can take advantage of lakeFS consistency guarantees and [CI/CD capabilities](../data_lifecycle_management) when ingesting data to S3 using lakeFS:
```
Comment (nopcoder) [inline docs/integrations/airbyte.md:32]: Update the title of the link 'CI/CD...' to match the page, "Data Lifecycle Management"

File: docs/integrations/airbyte.md [hunk 2]
```diff
@@ -9,9 +9,8 @@ has_children: false
 
 ---
 **Note:**
-Currently, Airbyte and lakeFS integration is broken on the latest Airbyte version ([link to issue #11871](https://github.com/airbytehq/airbyte/issues/11871)).  
-Until this issue is resolved, use the last known working version of Airbyte: [`v0.40.2`](https://github.com/airbytehq/airbyte/releases/tag/v0.40.2)
-Please note that this also affects the use of lakeFS and Airbyte Cloud.
+If using Airbyte OSS, please ensure you are using S3 destination connector version 0.3.17 or higher.
```
Comment (nopcoder) [inline docs/integrations/airbyte.md:12]: Link to the one we like the user to use (https://airbyte.com/connectors/s3)

**Review by nopcoder — CHANGES_REQUESTED** [review]

**Review by vinodhini8694 — APPROVED** [review]
LGTM

**Review by nopcoder — APPROVED** [review]
lgtm!

**Pull request merged**